src/crimson/net/SocketConnection.cc

```diff
@@ -515,16 +528,19 @@ seastar::future<>
 SocketConnection::send_connect_reply_ready(msgr_tag_t tag,
                                            bufferlist&& authorizer_reply)
 {
+  h.global_seq = get_messenger()->get_global_seq();
   h.reply.tag = tag;
   h.reply.features = policy.features_supported;
-  h.reply.global_seq = get_messenger()->get_global_seq();
+  h.reply.global_seq = h.global_seq;
   h.reply.connect_seq = h.connect_seq;
   h.reply.flags = 0;
   if (policy.lossy) {
     h.reply.flags = h.reply.flags | CEPH_MSG_CONNECT_LOSSY;
   }
   h.reply.authorizer_len = authorizer_reply.length();
-  return out.write(reinterpret_cast<const char*>(&h.reply), sizeof(h.reply))
+  seastar::net::packet msg{reinterpret_cast<const char*>(&h.reply),
+                           sizeof(h.reply)};
+  return out.write(std::move(msg))
```

Comment:
this code gets duplicated a bit - can we make a helper function for it?
```c++
template <typename T>
seastar::net::packet make_static_packet(const T& value) {
  return { reinterpret_cast<const char*>(&value), sizeof(value)) };
}
```
then these calls could just be `return out.write(make_static_packet(h.reply));`

Reply:
repushed and fixed